src/Knp/Menu/Twig/MenuExtension.php

```diff
@@ -3,27 +3,49 @@
 namespace Knp\Menu\Twig;
 
 use Knp\Menu\ItemInterface;
+use Knp\Menu\Util\MenuManipulator;
+use Knp\Menu\Matcher\MatcherInterface;
 
 class MenuExtension extends \Twig_Extension
 {
     private $helper;
+    private $matcher;
+    private $menuManipulator;
 
     /**
      * @param Helper $helper
      */
-    public function __construct(Helper $helper)
+    public function __construct(Helper $helper, MatcherInterface $matcher, MenuManipulator $menuManipulator)
```

Comment:
these dependencies should be optional (as done in the Helper) to maintain backward compatibility